bash_completion.sh

```diff
@@ -128,11 +128,36 @@ _buildtest ()
 {
   local cur="${COMP_WORDS[COMP_CWORD]}"
   local prev="${COMP_WORDS[COMP_CWORD-1]}"
+  declare -i offset="0"
 
   COMPREPLY=()   # Array variable storing the possible completions.
 
-  next=${COMP_WORDS[1]}
+  declare -a buildtest_opts=("--color" "--config" "--debug" "--editor" "--help" "--helpcolor" "--logpath" "--loglevel" "--print-log" "--no-color" "--report" "--version" "--view-log" "-c" "-d" "-h" "-l" "-p" "-r" "-V")
 
+  commands_with_input=( "--color" "--config" "-c" "--report" "-r" "--loglevel" "-l" "--editor" )   # Array variable storing commands which require an input argument from the user.
+
+  for command in "${COMP_WORDS[@]}"
+  do
+    for element in "${buildtest_opts[@]}"
+    do
+
+        if [[ "$command" == "$element" ]]; then
+          
+          if [[ "${commands_with_input[*]}" =~ $command ]];
+          then
+            ((offset+=2))
+          else
+            ((offset+=1))
+          fi
+        fi
+
+    done
+
+  done
+
+  local next=${COMP_WORDS[1+offset]}
+  local offset_by_two=2+offset
+  local offset_by_three=3+offset
```

Comment:
i find this bit confusing having variables `offset_by_two` and `offset_by_tree` since this can lead to endless amount of local variables.